extlibs/headers/drm/drm-common.c

```diff
@@ -285,34 +285,6 @@ int init_drm(struct drm *drm, const char *device, const char *mode_str,
 			printf("requested mode not found, using default mode!\n");
 	}
 
-	/* find preferred mode or the highest resolution mode: */
-	if (!drm->mode)
-	{
-		for (i = 0, area = 0; i < connector->count_modes; i++)
-		{
-			drmModeModeInfo *current_mode = &connector->modes[i];
-
-			if (current_mode->type & DRM_MODE_TYPE_PREFERRED)
-			{
-				drm->mode = current_mode;
-				break;
-			}
-
-			int current_area = current_mode->hdisplay * current_mode->vdisplay;
-			if (current_area > area)
-			{
-				drm->mode = current_mode;
-				area = current_area;
-			}
-		}
-	}
-
-	if (!drm->mode)
-	{
-		printf("could not find mode!\n");
-		return -1;
-	}
-
```

Comment:
We was under the impression that this is essentially a library. If we have to adjust this on our own, I feel like the functionality should not be part of a file in the extlib folder, but instead be part of SFML itself. 🤔

Reply:
It's technically just a set of common functions written by Rob Clark, taken from kmscube example. I doubt it's still actively maintained. Btw, I thought we settled on it sitting inside the extlibs folder. Does the fact of it sitting there prevent us from improving it?

Reply:
I remember bringing this up. If we are to add 3rd party sources to the src/ directory then we need to (1) make sure the licensing allows that, (2) wholly change the code to match SFML conventions and standards, and (3) be willing to maintain that code for perpetuity. I don't think this code checked off all 3 requirements so it got relegated to extlibs/. The assumption then becomes that because we put the code in extlibs/, we're never going to modify it since it's not our code and only lives in our repo because we couldn't be bothered to consume it from some external source. 

We're now in this ugly middle ground where we have 3rd party that we don't actually want to maintain but do need to modify and I'm not sure how to square those two facts.

Reply:
Me neither, any volunteers to SFML it ;) ?

Reply:
Whoever is most invested in the DRM backend ought to volunteer and SFML-itize this code. Happy to help but I can't lead that effort. At a minimum, let's make sure there are no licensing concerns. @oomek can you look into the original license of this code lets us totally take and rewrite it?

Reply:
https://github.com/freedesktop/mesa-kmscube/blob/master/common.c

As long as the header is intact we can modify it.

Reply:
The copyright header? Or the C++ header?

Reply:
Copyright header

Reply:
Here is the copyright note from the original repo, the github one is just a mirror
https://gitlab.freedesktop.org/mesa/kmscube/-/blob/master/COPYING

Reply:
We can store both files inside src, up to you

Reply:
As in, does the 3rd party DRM header _need_ to be part of SFML's public interface or is merely an implementation detail?

Reply:
Not sure if it's required to add it to public interface it's referenced just by
`src/SFML/Window/DRM/VideoModeImpl.cpp`
`src/SFML/Window/DRM/WindowImplDRM.cpp`
`src/SFML/Window/DRM/DRMContext.hpp`

What if we take it over, host on our repo, and maintain it as an extlib? From what I read the licence allows it.

Reply:
It sounds like SFML-itizing these 2 files and keeping them as implementation details in src/ is viable. I'm curious what the rest of the team thinks about this though.

Reply:
I was under the impression that it's a file/library that "everyone is using", and thus thought it was unmodified or only slightly wrapped. If we can't use it as quasi library, then we should move the functionality into our own classes and structures, meaning the goal shouldn't necessarily be to have a `src/SFML/Window/DRM/Common.cpp` but instead integrate the functionality in the existing Impl and Context classes.

Reply:
Whatever makes you happy. Shouldn't we delegate this task to a separate PR then?

Reply:
> I was under the impression that it's a file/library that "everyone is using", and thus thought it was unmodified or only slightly wrapped.

This was my impression too.

> Shouldn't we delegate this task to a separate PR then?

I think this reorganization should be its own PR before we start modifying the code from how we originally got it, but I still want some more of the team to weigh in on this.

Reply:
https://github.com/ChrisThrasher/SFML/actions/runs/3267610846/jobs/5373007062

The existence of 3rd party source files in our build is getting in the way of using clang-tidy. I could probably figure out a workaround but would prefer not to. Add that to the list of reasons we ought to integrate this file into SFML itself.

@oomek Can you try integrating these files into SFML itself in a separate branch not including any of this PR's changes? At a minimum, move the files to an appropriate place in src/ and modify the build system to treat them as 1st party files. Presumably that won't be too hard and then we can go from there determining whether we want to start heavily modifying the files to look like 1st party SFML code.

Reply:
@ChrisThrasher This commit targets 2.6.X Can we open a separate issue for this please? It's getting really messy here.